armi/reactor/tests/test_assemblies.py

```diff
@@ -1022,10 +1022,31 @@ def test_rotate(self):
         rotY = -0.5
         a.add(b)
         a.rotate(math.radians(120))
+        # test list rotation
         self.assertEqual(a.getBlocks()[0].p.THcornTemp, rotTemp)
         self.assertAlmostEqual(a.getBlocks()[0].p.displacementX, rotX)
         self.assertAlmostEqual(a.getBlocks()[0].p.displacementY, rotY)
 
+        b.p.THcornTemp = np.array([400, 450, 500, 550, 600, 650])
+        rotTemp = np.array([600, 650, 400, 450, 500, 550])
+        a.rotate(math.radians(120))
+        # test np.ndarray rotation
+        for i in range(len(b.p.THcornTemp)):
+            self.assertEqual(a.getBlocks()[0].p.THcornTemp[i], rotTemp[i])
+
+        # test that floats and ints are left alone
+        b.p.THcornTemp = 3
+        a.rotate(math.radians(120))
+        self.assertEqual(a.getBlocks()[0].p.THcornTemp, 3)
+        b.p.THcornTemp = 4.0
+        a.rotate(math.radians(120))
+        self.assertEqual(a.getBlocks()[0].p.THcornTemp, 4.0)
+
+        # check that TypeError is raised for unexpected data type
+        b.p.THcornTemp = "bad data"
+        with self.assertRaises(TypeError):
+            a.rotate(math.radians(120))
+
```

Comment:
I'm not sure if rotatePins is tested somewhere, but it might be good to test that as well. Also, you may want to add a test for the list equivalent since it looks like the rotate method works on lists or numpy arrays of length 6. Could you also add a test that fails to change a list of length != 6 and same for a numpy arrays?

Reply:
`rotatePins` does have a unit test already

@jasonbmeng for writing tests where you need to test the output of the `runLog` like Jake is suggesting you would do the following:

```python
from armi.tests import mockRunLogs

b.p.THcornTemp = np.array([400, 450, 500]) 
with mockRunLogs.BufferLog() as mock:
    self.assertEqual("", mock.getStdout())
    a.rotate(math.radians(120))
    self.assertIn("No rotation performed", mock.getStdout())

# repeat this for a list with len 1-5 or > 6 too
```

Reply:
Thanks for the feedback @jakehader 
@opotowsky mentioned there is already testing for `rotatePins`
The original unit test for `b.rotate()` has testing for lists.

Thanks for the mockRunLog info. I pushed a commit adding testing for lists of len 5 and 7, and ndarrays of len 5 and 7.